spec/features/profile_spec.rb

```diff
@@ -95,6 +95,38 @@ def mock_writing_to_filesystem
   end
 end
 
+RSpec.feature "Visit page with special role", :type => :feature do
+  before :each do
+    @profile = FactoryGirl.create(:profile)
+    @user = FactoryGirl.create(:user, role: :pupil, profile: @profile)
+    @event = FactoryGirl.create(:event)
+    @application_letter = FactoryGirl.create(:application_letter_accepted, user: @user, event: @event)
+    login_as(@user, scope: :user)
+    visit profile_path(@user.profile)
```

Comment:
I think you can drop this line and the three lines above.
You dont use events/application letters here, and since you are testing with different roles you need to re-login and visit each at the beginning of each test anyway, so we dont need it here